services:
  portainer:
    container_name: portainer
    image: portainer/portainer-ce:2.33.1-alpine@sha256:38f6d57dad1d47ad095f9d052f4dbae8551ece9358c6c0b9696770c42327d66d
    restart: unless-stopped
    environment:
      - TZ=America/New_York
    networks:
      - proxy
    volumes:
      - '/var/run/docker.sock:/var/run/docker.sock'
      - 'portainer_data:/data'
    labels:
      - traefik.enable=true
      - traefik.http.routers.portainer.rule=Host(`portainer.${SECRET_INTERNAL_DOMAIN_NAME}`)
      - traefik.http.routers.portainer.tls=true
      - traefik.http.routers.portainer.tls.certresolver=cloudflare
      - traefik.http.routers.portainer.entrypoints=websecure
      - traefik.http.services.portainer.loadbalancer.server.scheme=https
      - traefik.http.services.portainer.loadbalancer.server.port=9443

networks:
  proxy:
    external: true

volumes:
  portainer_data:
    driver: local
